tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -447,7 +502,10 @@ Status TRTEngineOp::GetEngineInputShapes(
     const CacheType& cache, const std::vector<TensorShape>& actual_input_shapes,
     std::vector<TensorShape>* engine_input_shapes) {
   // VerifyInputShapes() already ensured that all input shapes have same
-  // batch size, and are not scalars.
+  // batch size, and are not scalars, if we are in implicit batch mode.
+  //
+  // In explicit batch mode we plan to have single engine in the cache, and we
+  // return with its shape if it is compatible.
```

Comment:
This routine finds the engines with input shapes compatible with the actual_input_shapes, and returns the input shapes of  one of such engines that has the smallest batch size. I don't feed the old and new comment here are very helpful.